### DISCUSSION TITLE
Long type point id overflow?

### DISCUSSION THREAD

**Discussion by xtyDoge** [qdrant/qdrant#2560] [Q&A] [answered]
I insert a point with long type id `618919828488457344`
and when I query, the id returned becomes `618919828488457300`
The id is generated by snowflake id generator, and I store it as Long type in Java, but it seems to lose precision there.
Is that a bug?

**Comment by generall** [comment]
What type of client are you using? Usually it's related to JS limitations

**Reply by timvisee** [reply]
I've just confirmed that this ID does indeed work fine with manual REST requests.

**Reply by timvisee** [reply, marked as answer]
IIRC, the highest `Number` JS can hold without rounding is 4503599627370495 (2^52-1). That's why you might see the last two digits in your ID being rounded down.

**Reply by xtyDoge** [reply]
U get the point, when I use curl command instead of web ui, it shows the correct id. Thx
